homework/hw10_regexp.py

```diff
@@ -0,0 +1,30 @@
+"""
+HOMEWORK10_Regexp.
+
+Task Description:
+-----------------
+HOMEWORK10 involves the following task.
+Open input.txt file and find all small english letters that match conditions:
+Target small letter round exactly with three capital english letters
+on the left and on the right.
+Examples:
+
+sdTRYaUVKn -> matches "a" because 'TRY' on the left (3 capital letters)
+and 'UVK' on the right (3 capital letters)
+
+NTRSjARTb -> no match
+(not exactly 3 capital letters on the left ('NTRS' = 4 letters))
+
+zDFGbOPNq -> matches "b"
+
+Print Output as : "Result: "string of found letters">
+"""
+
+import re
+
+with open("input.txt", "r") as input_file:
+    content = input_file.read()
+
+characters = re.findall(r'(?<![A-Z])[A-Z]{3}([a-z])[A-Z]{3}(?![A-Z])', content)
```

Comment:
Yes acceptable and will give correct result , as advanced just pay attention that we could build regexp without groups even - just using PRE and POST in correct way
`match_pattern = r'(?<=[^A-Z][A-Z]{3})[a-z](?=[A-Z]{3}[^A-Z])'`